sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/implementation/converters/CommunicationIdentifierConverter.java

```diff
@@ -32,31 +33,37 @@ public static CommunicationIdentifier convert(CommunicationIdentifierModel ident
 
         assertSingleType(identifier);
         String rawId = identifier.getRawId();
+        CommunicationIdentifierModelKind kind = (identifier.getKind() != null) ? identifier.getKind() : extractKind(identifier);
 
-        if (identifier.getCommunicationUser() != null) {
-            Objects.requireNonNull(identifier.getCommunicationUser().getId());
+        if (kind == CommunicationIdentifierModelKind.COMMUNICATION_USER
+            && identifier.getCommunicationUser() != null) {
+            Objects.requireNonNull(identifier.getCommunicationUser().getId(),
+                "'ID' of the CommunicationUserIdentifierModel cannot be null.");
             return new CommunicationUserIdentifier(identifier.getCommunicationUser().getId());
         }
 
-        if (identifier.getPhoneNumber() != null) {
-            PhoneNumberIdentifierModel phoneNumberModel = identifier.getPhoneNumber();
-            Objects.requireNonNull(phoneNumberModel.getValue());
-            return new PhoneNumberIdentifier(phoneNumberModel.getValue()).setRawId(rawId);
+        if (kind == CommunicationIdentifierModelKind.PHONE_NUMBER
+            && identifier.getPhoneNumber() != null) {
+            String phoneNumber = identifier.getPhoneNumber().getValue();
+            Objects.requireNonNull(phoneNumber, "'PhoneNumber' of the CommunicationUserIdentifierModel cannot be null.");
```

Comment:
`CommunicationUserIdentifierModel` doesn't contain a `PhoneNumber`. It should either be `CommunicationIdentifierModel`, or its submodel `PhoneNumberIdentifierModel`.